domains/client/domain-executor/src/tests.rs

```diff
@@ -386,140 +398,112 @@ async fn pallet_domains_unsigned_extrinsics_should_work() {
-                    / 2000) as u64,
-                extrinsics_root: Default::default(),
-            },
-            receipts: vec![execution_receipt],
-            extrinsics: Vec::<UncheckedExtrinsic>::new(),
-        };
-
-        let pair = ExecutorPair::from_string("//Alice", None).unwrap();
-        let signature = pair.sign(bundle.hash().as_ref());
-
-        let signed_opaque_bundle = SignedBundle {
-            bundle,
-            bundle_solution: BundleSolution::System {
-                authority_stake_weight: Default::default(),
-                authority_witness: Default::default(),
-                proof_of_election: ProofOfElection::dummy(DomainId::SYSTEM, pair.public()),
-            }, // TODO: mock ProofOfElection properly
-            signature,
-        }
-        .into_signed_opaque_bundle();
+    futures::join!(alice.wait_for_blocks(5), ferdie.produce_blocks(5))
+        .1
+        .unwrap();
 
-        let tx = subspace_test_runtime::UncheckedExtrinsic::new_unsigned(
-            pallet_domains::Call::submit_bundle {
-                signed_opaque_bundle,
+    let ferdie_client = ferdie.client.clone();
+    let construct_submit_bundle =
+        |primary_number: BlockNumber| -> subspace_test_runtime::UncheckedExtrinsic {
+            let execution_receipt = crate::aux_schema::load_execution_receipt(
+                &*alice.backend,
+                alice.client.hash(primary_number).unwrap().unwrap(),
+            )
+            .expect("Failed to load execution receipt from the local aux_db")
+            .unwrap_or_else(|| {
+                panic!("The requested execution receipt for block {primary_number} does not exist")
+            });
+
+            let bundle = Bundle {
+                header: BundleHeader {
+                    primary_number,
+                    primary_hash: ferdie_client.hash(primary_number).unwrap().unwrap(),
+                    slot_number: (std::time::SystemTime::now()
+                        .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                        .expect("Current time is always after unix epoch; qed")
+                        .as_millis()
+                        / 2000) as u64,
+                    extrinsics_root: Default::default(),
+                },
+                receipts: vec![execution_receipt],
+                extrinsics: Vec::<UncheckedExtrinsic>::new(),
+            };
+
+            let pair = ExecutorPair::from_string("//Alice", None).unwrap();
+            let signature = pair.sign(bundle.hash().as_ref());
+
+            let signed_opaque_bundle = SignedBundle {
+                bundle,
+                bundle_solution: BundleSolution::System {
+                    authority_stake_weight: Default::default(),
+                    authority_witness: Default::default(),
+                    proof_of_election: ProofOfElection::dummy(DomainId::SYSTEM, pair.public()),
+                }, // TODO: mock ProofOfElection properly
```

Comment:
Why not adjust the API `produce_slot_and_wait_for_bundle_submission` to return the produced bundle as well so that it don't have constructe it manually and have a proper bundle solution?

Reply:
The test `pallet_domains_unsigned_extrinsics_should_work` has been rewritten according to the comment and added more cases, the changes have been put to the original commit, PTAL.